pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -0,0 +1,463 @@
+			  "num_resolved": "0",
+			  "resolved": ""
+			}
+		  },
+		  "client": "Grafana",
+		  "client_url": "http://localhost:3000/",
+		  "links": [
+			{
+			  "href": "http://localhost:3000/",
+			  "text": "External URL"
+			}
+		  ]
+		}`,
+	},
+}
+
+func getAlertmanagerConfig(channelAddr string) string {
+	return strings.ReplaceAll(alertmanagerConfig, "CHANNEL_ADDR", channelAddr)
+}
+
+func getRulesConfig(t *testing.T) string {
+	interval, err := model.ParseDuration("10s")
+	require.NoError(t, err)
+	rules := apimodels.PostableRuleGroupConfig{
+		Name:     "arulegroup",
+		Interval: interval,
+	}
+
+	// Create rules that will fire as quickly as possible for all the routes.
+	for _, alertName := range alertNames {
+		rules.Rules = append(rules.Rules, apimodels.PostableExtendedRuleNode{
+			GrafanaManagedAlert: &apimodels.PostableGrafanaRule{
+				Title:     alertName,
+				Condition: "A",
+				Data: []ngmodels.AlertQuery{
+					{
+						RefID: "A",
+						RelativeTimeRange: ngmodels.RelativeTimeRange{
+							From: ngmodels.Duration(time.Duration(5) * time.Hour),
+							To:   ngmodels.Duration(time.Duration(3) * time.Hour),
+						},
+						DatasourceUID: "-100",
+						Model: json.RawMessage(`{
+							"type": "math",
+							"expression": "2 + 3 > 1"
+						}`),
+					},
+				},
+			},
+		})
+	}
+
+	b, err := json.Marshal(rules)
+	require.NoError(t, err)
+
+	return string(b)
+}
+
+func TestNotificationChannels(t *testing.T) {
+	dir, path := testinfra.CreateGrafDir(t, testinfra.GrafanaOpts{
+		EnableFeatureToggles: []string{"ngalert"},
+		AnonymousUserRole:    models.ROLE_EDITOR,
+	})
+
+	store := testinfra.SetUpDatabase(t, dir)
+	grafanaListedAddr := testinfra.StartGrafana(t, dir, path, store)
+
+	mockChannel := newMockNotificationChannel(t, grafanaListedAddr)
+	amConfig := getAlertmanagerConfig(mockChannel.server.Addr)
+	rulesConfig := getRulesConfig(t)
+
+	channels.SlackAPIEndpoint = fmt.Sprintf("http://%s/slack_recvX/slack_testX", mockChannel.server.Addr)
+	channels.PagerdutyEventAPIURL = fmt.Sprintf("http://%s/pagerduty_recvX/pagerduty_testX", mockChannel.server.Addr)
+
+	// There are no notification channel config initially.
+	alertsURL := fmt.Sprintf("http://%s/api/alertmanager/grafana/config/api/v1/alerts", grafanaListedAddr)
+	_ = getRequest(t, alertsURL, http.StatusNotFound)
+
+	// Create the namespace we'll save our alerts to.
+	require.NoError(t, createFolder(t, store, 0, "default"))
+
+	// Post the alertmanager config.
+	u := fmt.Sprintf("http://%s/api/alertmanager/grafana/config/api/v1/alerts", grafanaListedAddr)
+	postRequest(t, u, amConfig, http.StatusAccepted)
+
+	// Verifying that all the receivers and routes have been registered.
+	resp := getRequest(t, alertsURL, http.StatusOK)
+	b, err := ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+
+	cfg := apimodels.GettableUserConfig{}
+	require.NoError(t, json.Unmarshal(b, &cfg))
+
+	require.Equal(t, totalReceivers, len(cfg.AlertmanagerConfig.Receivers))
```

Comment:
I'd like us to assert the actual payload here if possible, I'm certain there are a few things we'll need to fix as part of this PR once we do.